gradio/helpers.py

```diff
@@ -797,3 +806,24 @@ def get_color_gradient(c1, c2, n):
 
     subprocess.call(ffmpeg_cmd, shell=True)
     return output_mp4.name
+
+
+@document()
+class EventData:
+    """
+    When a subclass of EventData is added as a type hint to an argument of an event listener method, this object will be passed into that method.
+    It contains information about the event that triggered the listener, including the target object and any other optional data.
+
+    Example:
+        gallery = gr.Gallery()
+        focused = gr.Textbox()
+
+        def on_focus(evt: gr.SelectData):  # SelectData is a subclass of EventData
+            return evt.index
+
+        gallery.select(on_focus, None, focused)
+    """
+
+    def __init__(self, target: Block | None, data: Any):
+        self.target = target
```

Comment:
Is target used anywhere yet? We should add an example and explain what it can be used for!

Reply:
added to example code snippet